File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/inscriptionref/InviteStudentCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.inscriptionref;

import java.util.List;
import ma.zs.alc.bean.core.inscriptionref.InviteStudent;
import ma.zs.alc.dao.criteria.core.inscriptionref.InviteStudentCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface InviteStudentCollaboratorService extends  IService<InviteStudent,InviteStudentCriteria>  {

    List<InviteStudent> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);




}

File: frontend-bqckup/src/app/shared/service/collaborator/inscriptionref/InviteStudentCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {InviteStudentDto} from '../../../model/inscriptionref/InviteStudent.model';
import {InviteStudentCriteria} from '../../../criteria/inscriptionref/InviteStudentCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class InviteStudentCollaboratorService extends AbstractService<InviteStudentDto, InviteStudentCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/inviteStudent/';
    }

    public constrcutDto(): InviteStudentDto {
        return new InviteStudentDto();
    }

    public constrcutCriteria(): InviteStudentCriteria {
        return new InviteStudentCriteria();
    }
}
